universalmutator/mutator.py

```diff
@@ -69,12 +75,36 @@ def compileRules(ruleFiles):
     return (rules, ignoreRules, skipRules)
 
 
+def mutants_comby(source, ruleFiles=["universal.rules"], mutateTestCode=False, mutateBoth=False,
+            ignorePatterns=None, ignoreStringOnly=False, fuzzing=False, language=".generic"):
+    comby = Comby()
+    print("MUTATING WITH RULES (COMBY):", ", ".join(ruleFiles))
+    (rules, ignoreRules, skipRules) = compileRules(ruleFiles, True)
+    for p in ignorePatterns:
+        ignoreRules.append(lhs)
+    source = ''.join(source)
+    mutants = []
+    # Instead of line-by-line x rule-by-rule iterate rule-by-rule x match-by-match.
+    for ((lhs, rhs), ruleUsed) in rules:
+        try: 
+            for match in comby.matches(source, lhs, language=language):
```

Comment:
I think language=language is needed here since comby.matches takes 4 parameters and the 4th argument is language. If this is not written in this notation, language would be inferred to be the third parameter in comby.matches.